### PR TITLE
Hotfix

### PR DESCRIPTION
**Pull request by victor-paumier**
- fix sort by semesters on matching list
- Allow matching of mentee staying for a year with a mentor only available for a semester
- set default nb elem in lists to 10
- fix deleting thread participant on change user role
- add column visibly button on lists to hide columns dynamically
- popup list buddies on matching summary

### CODE REVIEW

File: assets/js/matching_summary.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import tools from './tools/tools';
+import Routing from './tools/Routing';
+
+$(document).ready(() => {
+    tools.initPopupOnClick($('.list-buddies_link'), 'user_buddies', Translator.trans('buddy.list.title'));
+    tools.initPopupOnClick($('.list-buddies-refused_link'), 'user_buddies_refused', Translator.trans('buddy.refused.list.title'));
+    tools.initPopupOnClick($('.list-reports_link'), 'buddy_report', Translator.trans('buddy.popup.single.title'));
+
+    $('#popup').on('shown.bs.modal', () => {
+        const buddiesList = '#list_buddies_user';
```
Comment (JeremieSamson) [inline assets/js/matching_summary.js]: why not `const $buddiesList = $('#list_buddies_user');` ?

File: src/Repository/UserRepository.php [hunk 2]
```diff
@@ -185,6 +159,38 @@ public function getUsersBySection(Section $section, bool $isLocal = null): array
         return $qb->getQuery()->getResult(Query::HYDRATE_SIMPLEOBJECT);
     }
 
+    public function getCurrentBuddies(
+        User $user,
+        bool $isLocal,
+        array $status = [
+            BuddyStatusEnum::STATUS_PENDING,
+        BuddyStatusEnum::STATUS_CONFIRMED,
```
Comment (JeremieSamson) [inline src/Repository/UserRepository.php]:
```suggestion
            BuddyStatusEnum::STATUS_CONFIRMED,
```

File: templates/Buddy/list.html.twig [hunk 3]
```diff
@@ -0,0 +1,74 @@
+<p>
+    {{ 'user.current'|trans ~ ': ' }}
+    <a href="mailto:{{ user.email }}">{{ user.email }}</a>
+</p>
+
```
Comment (JeremieSamson) [inline templates/Buddy/list.html.twig]: remove line plz

File: src/Repository/UserRepository.php (hunk 2 above)
Comment (victor-paumier) [reply src/Repository/UserRepository.php]: impossible: cs-fixer

File: assets/js/matching_summary.js (hunk 1 above)
Comment (victor-paumier) [reply assets/js/matching_summary.js]: I need `'#list_buddies_user'` l.18

Comment (JeremieSamson) [reply assets/js/matching_summary.js]: didn't see where it is used ?

File: src/Repository/UserRepository.php (hunk 2 above)
Comment (JeremieSamson) [reply src/Repository/UserRepository.php]: As we spoke, can you check which rule of php cs say ti write it like that ?

File: assets/js/matching_summary.js (hunk 1 above)
Comment (victor-paumier) [reply assets/js/matching_summary.js]: $(`${buddiesList} tbody tr`) (Line 18)

File: src/Repository/UserRepository.php (hunk 2 above)
Comment (victor-paumier) [reply src/Repository/UserRepository.php]: PedroTroller/line_break_between_method_arguments

**Pull request merged**